sqlmesh/core/loader.py

```diff
@@ -617,19 +620,29 @@ def process_models(completed_model: t.Optional[Model] = None) -> None:
             if not deps:
                 del graph[name]
                 model = models[name]
-                model.update_schema(schema)
-                futures.add(executor.submit(load_optimized_query_cache, model))
+                futures.add(
+                    executor.submit(
+                        load_optimized_query_and_mapping,
+                        model,
+                        mapping={
+                            parent: models[parent].columns_to_types
+                            for parent in model.depends_on
+                            if parent in models
+                        },
+                    )
+                )
 
-    with optimized_query_cache_pool(optimized_query_cache) as executor:
+    with optimized_query_cache_pool(optimized_query_cache, audits) as executor:
         process_models()
 
         while futures:
             for future in as_completed(futures):
                 futures.remove(future)
-                fqn, entry_name = future.result()
+                fqn, entry_name, data_hash, metadata_hash, mapping_schema = future.result()
                 model = models[fqn]
-                if entry_name:
-                    optimized_query_cache.with_optimized_query(model, entry_name)
-
+                model._data_hash = data_hash
+                model._metadata_hash = metadata_hash
+                model.update_schema(mapping_schema)
```

Comment:
We wouldn't have to do this twice, if we were updating the schema prior to submitting a task like we used to

Reply:
we need to do it here on the computed mapping because it was done in a different process. we do it in the process so that we can leverage cores